app/controllers/publishers/omniauth_callbacks_controller.rb

```diff
@@ -21,17 +21,12 @@ def register_youtube_channel
       existing_channel = Channel.joins(:youtube_channel_details).
           where("youtube_channel_details.youtube_channel_id": youtube_channel_data['id']).first
 
-      if existing_channel
-        if existing_channel.publisher == current_publisher
-          redirect_to home_publishers_path, notice: t(".channel_already_registered")
-          return
-        else
-          redirect_to home_publishers_path, flash: { taken_channel_id: existing_channel.id }
-          return
-        end
+      if existing_channel && existing_channel.publisher == current_publisher
```

Comment:
We could use the [nil-safe navigation operator](http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/) here which would look like
`if existing_channel&.publisher == current_publisher`